core/src/main/java/hudson/model/Computer.java

```diff
@@ -1467,6 +1467,11 @@ public void doConfigSubmit( StaplerRequest req, StaplerResponse rsp ) throws IOE
             throw new FormException(Messages.ComputerSet_SlaveAlreadyExists(proposedName), "name");
         }
 
+        String nExecutors = req.getSubmittedForm().getString("numExecutors");
+        if (Integer.parseInt(nExecutors)<=0) {
+            throw new FormException(Messages.Slave_InvalidConfig_Executors(nodeName), "numExecutors");
```

Comment:
As below, this would more naturally live in `Slave.setNumExecutors` but that would be a source-incompatible change.